hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java

```diff
@@ -247,21 +354,72 @@ public void reinitialize(
     // before reinitialization. Just update deletedTable here.
     Preconditions.checkArgument(deletingTxIDs.isEmpty());
     this.deletedTable = deletedBlocksTXTable;
+    this.statefulConfigTable = configTable;
   }
 
   public static Builder newBuilder() {
     return new Builder();
   }
 
+  @Override
+  public DeletedBlocksTransactionSummary getTransactionSummary() {
+    return DeletedBlocksTransactionSummary.newBuilder()
+        .setFirstTxID(firstTxIdForDataDistribution)
+        .setTotalTransactionCount(totalTxCount.get())
+        .setTotalBlockCount(totalBlockCount.get())
+        .setTotalBlockSize(totalBlocksSize.get())
+        .setTotalBlockReplicatedSize(totalReplicatedBlocksSize.get())
+        .build();
+  }
+
+  private void initDataDistributionData() throws IOException {
+    if (VersionedDatanodeFeatures.isFinalized(HDDSLayoutFeature.DATA_DISTRIBUTION)) {
```

Comment:
we can keep populating memory information, no need feature flag as there is no impact.